From 97ac574274697b391728a88be98a5cc503340ecc Mon Sep 17 00:00:00 2001
From: Abhijith Das <adas@redhat.com>
Date: Thu, 17 Apr 2014 20:37:46 -0400
Subject: [fs] gfs2: Refresh quota after setting through quotactl

Message-id: <1397767066-7475-1-git-send-email-adas@redhat.com>
Patchwork-id: 78968
O-Subject: [RHEL7.1 PATCH] GFS2: Refresh quota after setting through quotactl
Bugzilla: 1088740
RH-Acked-by: Robert S Peterson <rpeterso@redhat.com>
RH-Acked-by: Steven Whitehouse <swhiteho@redhat.com>

This patch fixes a condition where a stale gfs2_quota_data was
being used instead of the most recently modified one.

This patch is already upstream:
http://git.kernel.org/cgit/linux/kernel/git/steve/gfs2-3.0-nmw.git/commit/fs/gfs2?id=991deec8193a20cf5a79a4c4bc01ba1a51c9ba5f

Resolves: rhbz#1088740
Signed-off-by: Abhi Das <adas@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/gfs2/quota.c b/fs/gfs2/quota.c
index 0c446c8..f279dfb 100644
--- a/fs/gfs2/quota.c
+++ b/fs/gfs2/quota.c
@@ -747,6 +747,7 @@ get_a_page:
   i_size_write(inode, size);
  inode->i_mtime = inode->i_atime = CURRENT_TIME;
  mark_inode_dirty(inode);
+ set_bit(QDF_REFRESH, &qd->qd_flags);
  return 0;
 
 unlock_out:
-- 
1.7.1